app/Http/Controllers/Auth/SocialiteController.php

```diff
@@ -159,42 +162,79 @@ private function setRolesFromClaim(string $provider, $user): bool
     {
         $scopes = LibrenmsConfig::get('auth.socialite.scopes');
         $claims = LibrenmsConfig::get('auth.socialite.claims');
+        $is_debug = LibrenmsConfig::get('auth.socialite.debug');
+        $monolog = null;
 
-        if (is_array($scopes) &&
-            $this->socialite_user instanceof \Laravel\Socialite\AbstractUser &&
-            ! empty($claims)
-        ) {
-            $roles = [];
-            $attributes = $this->socialite_user->getRaw();
-
-            if (is_object(current($attributes)) && method_exists(current($attributes), 'getName') && method_exists(current($attributes), 'getAllAttributeValues')) {
-                $parsed_attributes = [];
-                foreach ($attributes as $attribute_object) {
-                    $attribute_name = $attribute_object->getName();
-                    $attribute_values = $attribute_object->getAllAttributeValues();
-                    $parsed_attributes[$attribute_name] = $attribute_values;
-                }
-                $attributes = $parsed_attributes;
+        try {
+            if ($is_debug) {
+                /** @var \Monolog\Logger $monolog */
+                $monolog = Log::getLogger();
+                $debugHandler = new StreamHandler(LibrenmsConfig::get('log_dir') . '/auth.log', Level::Debug);
```

Comment:
This should probably be a log channel.  I had some log relat d changes and that moved towards that.

Reply:
#10559